src/Task.ts

```diff
@@ -259,159 +233,34 @@ export class Task {
-                // and it might appear in the next (earlier in the line) tokens to parse
-                recurrenceRule = recurrenceMatch[1].trim();
-                description = description.replace(TaskRegularExpressions.recurrenceRegex, '').trim();
-                matched = true;
-            }
-
-            // Match tags from the end to allow users to mix the various task components with
-            // tags. These tags will be added back to the description below
-            const tagsMatch = description.match(TaskRegularExpressions.hashTagsFromEnd);
-            if (tagsMatch != null) {
-                description = description.replace(TaskRegularExpressions.hashTagsFromEnd, '').trim();
-                matched = true;
-                const tagName = tagsMatch[0].trim();
-                // Adding to the left because the matching is done right-to-left
-                trailingTags = trailingTags.length > 0 ? [tagName, trailingTags].join(' ') : tagName;
-            }
-
-            runs++;
-        } while (matched && runs <= maxRuns);
-
-        // Now that we have all the task details, parse the recurrence rule if we found any
-        if (recurrenceRule.length > 0) {
-            recurrence = Recurrence.fromText({
-                recurrenceRuleText: recurrenceRule,
-                startDate,
-                scheduledDate,
-                dueDate,
-            });
+        const { taskSerializer } = getTaskFormat();
+        const taskInfo = taskSerializer.deserialize(description);
+        if (taskInfo == null) {
+            return null;
         }
 
+        let scheduledDateIsInferred = false;
         // Infer the scheduled date from the file name if not set explicitly
-        if (DateFallback.canApplyFallback({ startDate, scheduledDate, dueDate }) && fallbackDate !== null) {
-            scheduledDate = fallbackDate;
+        if (DateFallback.canApplyFallback(taskInfo) && fallbackDate !== null) {
+            taskInfo.scheduledDate = fallbackDate;
             scheduledDateIsInferred = true;
         }
 
-        // Add back any trailing tags to the description. We removed them so we can parse the rest of the
-        // components but now we want them back.
-        // The goal is for a task of them form 'Do something #tag1 (due) tomorrow #tag2 (start) today'
-        // to actually have the description 'Do something #tag1 #tag2'
-        if (trailingTags.length > 0) description += ' ' + trailingTags;
-
-        // Tags are found in the string and pulled out but not removed,
-        // so when returning the entire task it will match what the user
-        // entered.
-        // The global filter will be removed from the collection.
-        const hashTagMatch = description.match(TaskRegularExpressions.hashTags);
-        if (hashTagMatch !== null) {
-            tags = hashTagMatch.filter((tag) => tag !== globalFilter).map((tag) => tag.trim());
```

Comment:
@kedestin not starting a review here, just curious regarding this change
-`tags = hashTagMatch.filter((tag) => tag !== globalFilter).map((tag) => tag.trim());`
+`taskInfo.tags = taskInfo.tags.filter((tag) => tag !== globalFilter);`

I was playing with this code when exploring the global filter and my feeling is that removing the `trim()` would alter the content of the `tags` -> `taskInfo.tags`. The resulting `taskInfo.tags` will probably have empty trailing spaces...

Reply:
Ah so now `trim()` is actually called in `taskSerializer.deserialize()`, so before this line which is actually fixing one of the global filter issues we spotted with @claremacrae =) Thanks!

Reply:
I’m on mobile only so not keeping up with this, but if it’s a change in behaviour it would be helpful to have it protected by a test please.

Reply:
@kedestin if my understanding is correct, if `globalfilter = #task` then

before:
`- [ ] #task this is recognised by global filter`
`- [ ] and this is not #task`
`- [ ] and this #task neither`

after:
`- [ ] #task this is recognised by global filter`
`- [ ] and this one is ok now #task`
`- [ ] and this #task as well`

I would do a simple test on this but too short of time this week

Reply:
With both Tasks 1.25.0 and this new code, all 3 of those tasks are actually recognised as tasks, with `#task` as the global filter.

Reply:
@kedestin @claremacrae thanks both for clarification.

The results (`With both Tasks 1.25.0 and this new code, all 3 of those tasks are actually recognised as tasks`) are astonishing to me but this means I have been wrong on my assumption that the behaviour should change.